eventmesh-connectors/eventmesh-connector-http/src/main/java/org/apache/eventmesh/connector/http/sink/handle/WebhookHttpSinkHandler.java

```diff
@@ -84,33 +108,74 @@ private void doInitExportServer() {
             .method(HttpMethod.GET)
             .produces("application/json")
             .handler(ctx -> {
-                // get received data
-                Object data = this.receivedDataQueue.poll();
-                if (data != null) {
+                // Validate the request parameters
+                MultiMap params = ctx.request().params();
+                String pageNumStr = params.get(ParamEnum.PAGE_NUM.getValue());
+                String pageSizeStr = params.get(ParamEnum.PAGE_SIZE.getValue());
+                String type = params.get(ParamEnum.TYPE.getValue());
+
+                // 1. type must be "poll" or "peek" or null
+                // 2. if type is "peek", pageNum must be greater than 0
+                // 3. pageSize must be greater than 0
+                if ((type != null && !Objects.equals(type, TypeEnum.PEEK.getValue()) && !Objects.equals(type, TypeEnum.POLL.getValue()))
+                    || (Objects.equals(type, TypeEnum.PEEK.getValue()) && (StringUtils.isBlank(pageNumStr) || Integer.parseInt(pageNumStr) < 1))
+                    || (StringUtils.isBlank(pageSizeStr) || Integer.parseInt(pageSizeStr) < 1)) {
 
-                    // export the received data
+                    // Return 400 Bad Request if the request parameters are invalid
                     ctx.response()
                         .putHeader(HttpHeaders.CONTENT_TYPE, "application/json; charset=utf-8")
-                        .setStatusCode(HttpResponseStatus.OK.code())
-                        .send(JSONObject.of("data", data).toJSONString());
-                    if (log.isDebugEnabled()) {
-                        log.debug("Succeed to export callback data. Data: {}", data);
-                    } else {
-                        log.info("Succeed to export callback data.");
-                    }
-                } else {
-                    // no data to export
+                        .setStatusCode(HttpResponseStatus.BAD_REQUEST.code())
+                        .end();
+                    log.info("Invalid request parameters. pageNum: {}, pageSize: {}, type: {}", pageNumStr, pageSizeStr, type);
+                    return;
+                }
+
+                // Parse the request parameters
+                if (type == null) {
+                    type = TypeEnum.PEEK.getValue();
+                }
+                int pageNum = StringUtils.isBlank(pageNumStr) ? 1 : Integer.parseInt(pageNumStr);
+                int pageSize = Integer.parseInt(pageSizeStr);
```

Comment:
You can parse them where you get them from `MultiMap params`.

Reply:
However, if I directly parse, there might be exceptions like `NoPointException` that I cannot predict what parameters the user will pass. I can only safely parse after validating to ensure the parameters are legitimate.

Reply:
OK